### ISSUE TITLE
Issue on lambda argument of tess3 fucntion

### ISSUE THREAD

**Issue by jdalapicolla** [bcm-uga/TESS3_encho_sen#16]
Dear Team of BCM-UGA,

My name is Jeronymo and I am part of a research group in Brazil. We are trying to implement TESS3 to estimate the genetic structure using spatial information in our pipeline, following Caye et al. 2016 (Mol. Ecol. Res.), Caye et al. 2018 (The Ann. Appl. Studies), and the vignette (https://bcm-uga.github.io/TESS3_encho_sen/articles/main-vignette.html). We have two questions on TESS3 and its arguments:

1) On alpha parameter (regularization parameter) discussed in Caye et al 2016 and 2018, and implemented in the tess3() function as a 'lambda' argument. According to the papers "Large values ​​of alpha imply that ancestry coefficients have similar values ​​for nearby individuals, whereas small values ​​produce results close to STRUCTURE" (Caye et al 2016; page 542). We are testing different lambda values ​​to evaluate its effects in K estimation. We used lambda values ​​equal to 1, 5, 10, and 15. However, we would like to know if we would need to test much larger values, such as 50, 100, 1000. What range of values ​​do you consider as "small values" and "large values" to test alpha / lambda? Is there any indication of a range of values ​​to be tested?

2) Another fact is that comparing our preliminary results of TESS3 to our results from sNMF, there is a difference in the number of K, in Caye et al 2018, in some cases, TESS3 would be a better choice because it is model-free and it incorporates spatial information. In the case of disparate results for the number of K between sNMF and TESS3, which one should we choose? TESS following Caye et al. 2018 or does it depend on what we are trying to calculate and on the biological traits of the target species? Do you have any suggestions?

Thank you so much for your time,
Jeronymo Dalapicolla

**Comment by francoio** [comment]
hi

these are not real program issues but general statistical questions.  For 1), you could rely on cross-validation to determine acceptable lambda values. Run tess3 with the option "mask = 0.05" and then choose lambda values that lead to lower "entropy" score (object$crossvalid.crossentropy). The range of values to be tested is around 1. Larger values are meaningless. 

For 2), snmf is another model and its score is not exactly computed in the same way as in  tess3r (note: snmf is model-free too). So, do not compare tess3 and snmf on the basis of their 'crossentropy'. Perhaps the best way to choose lambda and K in a consistent way is to use tess3r only.  Generally including geographic coordinates provides additional power to detect clusters, so K could be bigger with non-zero lambda values. 

More importantly, the K values found with cross-entropy may be "statistically meaningful" values of K, but not necessarly the most interesting biologically.  (Think of hierarchical structure for example). So my advice would be to show the barplots for several Ks, and not decide which  K  is the best or the truth because it does not exist.  

olivier
